### PR TITLE
This PR is for issue #11155 "Keymanager APIs: gas limit api"

### PR DESCRIPTION
**Pull request by shenhanc**
This PR implements issue #11155 the first task "get gas limit for the validator key".

**What type of PR is this?**

> Feature

Implement: Keymanager APIs: get_limit api - subtask: "get gas limt"

### CODE REVIEW

File: proto/eth/service/key_management.proto [hunk 1]
```diff
@@ -167,6 +167,19 @@ service KeyManagement {
...
+
+  // GetGasLimit returns the execution gas limit for an individual validator.
+  //
+  // HTTP response status codes:
+  //  - 400: Bad request
+  //  - 401: Unauthorized
+  //  - 403: Forbidden
+  //  - 500: Validator internal error
+  rpc GetGasLimit(PubkeyRequest) returns (GetGasLimitResponse) {
+    option (google.api.http) = {
```
Comment (james-prysm) [inline proto/eth/service/key_management.proto:179]: you need to regenerate the proto using `./hack/update-go-pbs.sh` and updating the generated files associated to the proto. the proto itself doesn't contain the implementation of the function.

File: proto/eth/service/key_management.proto [hunk 2]
```diff
@@ -167,6 +167,19 @@ service KeyManagement {
...
+      get: "/internal/eth/v1/validator/{pubkey}/gaslimit"
```
Comment (james-prysm) [inline proto/eth/service/key_management.proto:180]: the associated endpoint factory `./validator/rpc/apimiddleware/endpoint_factory.go` will also need to be added for this feature to work.  once this is done it should work but will require curl or postman testing

File: proto/eth/service/key_management.proto (hunk 1 above)
Comment (shenhanc) [reply proto/eth/service/key_management.proto:179]: Thanks. Ran update-go-pbs.sh and only included key_management.pb* in the PR.

File: proto/eth/service/key_management.proto (hunk 2 above)
Comment (shenhanc) [reply proto/eth/service/key_management.proto:180]: Done, thanks.

Comment (james-prysm) [inline proto/eth/service/key_management.proto:180]: i also realized according to the spec the path should be gas_limit

File: validator/rpc/apimiddleware/endpoint_factory.go [hunk 3]
```diff
@@ -41,6 +41,8 @@ func (*ValidatorEndpointFactory) Create(path string) (*apimiddleware.Endpoint, e
 	case "/eth/v1/validator/{pubkey}/feerecipient":
 		endpoint.GetResponse = &getFeeRecipientByPubkeyResponseJson{}
 		endpoint.PostRequest = &setFeeRecipientByPubkeyRequestJson{}
+	case "/eth/v1/validator/{pubkey}/gaslimit":
```
Comment (james-prysm) [inline validator/rpc/apimiddleware/endpoint_factory.go:44]: you forgot to add it in the paths function. 
please try to curl request or postman script test this.

you can test the easiest by building the application from the cmd package
1. cd `cmd/validator`
2. `go build -o ./tmp/validator`
3. `./tmp/validator --web ---validators-external-signer-url=http://localhost:9000/` /*this triggers running the validator client in web3signer mode which does not require a prysm wallet*/
4. grab the bearer token and then you can test using postman or curl

File: proto/eth/service/key_management.proto (hunk 2 above)
Comment (shenhanc) [reply proto/eth/service/key_management.proto]: Ah, thanks for pointing out. My bad.

File: validator/rpc/apimiddleware/endpoint_factory.go (hunk 3 above)
Comment (shenhanc) [reply validator/rpc/apimiddleware/endpoint_factory.go]: Thanks for sharing the recipe, this makes things easier. Now following your recipe, I tested the server with:

curl  -H "Authorization: Bearer eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.e30.C-BuVfWhKpr9elB05GTJdEEx_8AzkImxzXL03IqcHR8" 127.0.0.1:7500/eth/v1/validator/0xb4844195ce8ca78d9d4f7ffdf4001cfdb079e059542bcc4f45ddfff2bcec7defb1482db4f9426f92f59972da395dd2b5/gas_limit

And got:
{"data":{"pubkey":"0xb4844195ce8ca78d9d4f7ffdf4001cfdb079e059542bcc4f45ddfff2bcec7defb1482db4f9426f92f59972da395dd2b5","gas_limit":"30000000"}}

**Review by james-prysm — APPROVED** [review]
Tested locally using Postman + config file and works well. Thanks for the contributions! i'll get this merged and mark the first of 3 apis done

**Pull request merged**